### PR TITLE
Correct handling of quorum timeout.

### PR DESCRIPTION
**Pull request by johnou**
if election times out the previous quorum was never cancelled..

### CODE REVIEW

File: protocols/raft/src/main/java/io/atomix/protocols/raft/roles/CandidateRole.java [hunk 1]
```diff
@@ -124,6 +124,7 @@ private void sendVoteRequests() {
         raft.transition(RaftServer.Role.FOLLOWER);
       }
     });
+    this.quorum = quorum;
```
Comment (johnou) [inline protocols/raft/src/main/java/io/atomix/protocols/raft/roles/CandidateRole.java]: quorum was never set so if electrion times out the previous quorum was never cancelled.

**Comment by johnou** [comment]
Don't merge yet, think I see a race condition.

File: protocols/raft/src/main/java/io/atomix/protocols/raft/roles/CandidateRole.java (hunk 1 above)
Comment (johnou) [reply protocols/raft/src/main/java/io/atomix/protocols/raft/roles/CandidateRole.java]: @kuujo not sure if this is quite right, wdyt?

**Comment by johnou** [comment]
@kuujo PTAL, hopefully this is what you had in mind.

**Comment by kuujo** [comment]
Restarting tests

**Pull request merged**